### ISSUE TITLE
Preserving whitespace between HTML Django blocks

### ISSUE THREAD

**Issue by muppetjones** [prettydiff/prettydiff#95]
I ran 'beautify' on two generic html django scripts, and the first looked pretty good, but the second eliminated the whitespace between block statements (which is odd because the first sample had some whitespace added).
(samples from [here](https://github.com/Glavin001/atom-beautify/issues/321#issuecomment-104395960))

#### First
```html
{% load staticfiles %}
<!DOCTYPE html>
<html lang="en">

<head>
    <title>To-do Lists</title>
    {# <link href="/static/semantic-ui/semantic.min.css" rel="stylesheet"> #}
    <link href="{% static 'semantic-ui/semantic.css' %}" type="text/css" rel="stylesheet">

</head>

<body>
    <div class="pusher">
        <div class="ui two column centered grid">
            <div class="column">
                <div class="ui center aligned basic segment">
                    <h1>{% block header_text %}{% endblock %}</h1>
                    <form method="POST" action="{% block form_action %}{% endblock %}">
                        {% csrf_token %}
                        <input name="item_text" id="id_new_todo_item" placeholder='Enter a to-do item' />
                    </form>
                </div>
                {% block table %} {% endblock %}
            </div>
        </div>
    </div>
</body>

</html>
```
to 
```html
{% load staticfiles %}
<!DOCTYPE html>
<html lang="en">
    <head>
        <title>To-do Lists</title>
        {# 
        <link href="/static/semantic-ui/semantic.min.css" rel="stylesheet">
        #} 
        <link href="{% static 'semantic-ui/semantic.css' %}" rel="stylesheet"
            type="text/css">
    </head>
    <body>
        <div class="pusher">
            <div class="ui two column centered grid">
                <div class="column">
                    <div class="ui center aligned basic segment">
                        <h1>{% block header_text %}{% endblock %}</h1>
                        <form action="{% block form_action %}{% endblock %}" method="POST">
                            {% csrf_token %}
                            <input id="id_new_todo_item" name="item_text" placeholder='Enter a to-do item'/>
                        </form>
                    </div>
                    {% block table %}
                    {% endblock %}
                </div>
            </div>
        </div>
    </body>
</html>
```

#### Second
```html
{% extends 'todo_lists/base.html' %}

{% block header_text %}Your to-do list{% endblock %}
{% block form_action %}/lists/{{ list.id }}/add_todo_item{% endblock %}

{% block table %}
    <table id="id_todo_list_table">
        {% for item in list.todoitem_set.all %}
        <tr>
            <td>{{ forloop.counter }}: {{ item.text }}</td>
        </tr>
        {% endfor %}
    </table>
{% endblock %}
```
to 
```html
{% extends 'todo_lists/base.html' %}
{% block header_text %}Your to-do list{% endblock %} {% block form_action
%}/lists/{{ list.id }}/add_todo_item{% endblock %} {% block table %}
<table id="id_todo_list_table">
    {% for item in list.todoitem_set.all %}
    <tr>
        <td>{{ forloop.counter }}: {{ item.text }}</td>
    </tr>
    {% endfor %}
</table>
{% endblock %}
```

**Comment by prettydiff** [comment]
I am not currently retaining blank lines in markup code.  All these various template schemes are a very new addition to a very old parser that I wrote. I am considering rewriting the markup parser to match the models used by my jspretty and csspretty libraries.

I do recognize that I am not currently supporting the `{# #}` delimiters.

I am going to close this bugs and recommend following these two:

* https://github.com/prettydiff/prettydiff/issues/93
* https://github.com/prettydiff/prettydiff/issues/87

**Comment by prettydiff** [comment]
Reopened because this is referenced by another bug in the Atom-Beautify repo: https://github.com/prettydiff/prettydiff/issues/87

**Comment by prettydiff** [comment]
A beta version of the new markup parser is up. This version is feature complete, but not fully tested against edge cases.

http://prettydiff.com/ignore/testlocation/prettydiff.com.xhtml

Please test this out and try to break it. If you encounter any issues please post them here.  This is likely to be buggy because it is completely new parser written from scratch in the past two weeks.

**Comment by prettydiff** [comment]
Fixed as v1.12.0
